shared/common-adapters/plain-input.desktop.js

```diff
@@ -0,0 +1,251 @@
+// @flow
+import * as React from 'react'
+import {getStyle as getTextStyle} from './text.desktop'
+import {collapseStyles, styleSheetCreate} from '../styles'
+
+import type {Props, Selection, TextInfo} from './plain-input'
+import {checkTextInfo} from './plain-input.shared'
+
+type State = {
+  value: string,
+}
+
+// A plain text input component. Handles callbacks, text styling, and auto resizing but
+// adds no styling
+class PlainInput extends React.PureComponent<Props, State> {
+  state: State
+  _input: HTMLTextAreaElement | HTMLInputElement | null
+  _isComposingIME: boolean = false
+
+  constructor(props: Props) {
+    super(props)
+    const value = props.value || ''
+    this.state = ({
+      value,
```

Comment:
as far as I can tell, in the original input component, we keep `state.value` around for the controlled case, so that we can compare the previous `state.value` with the current one and autoresize if necessary. It looks like we're getting rid of `componentDidUpdate` and the call to `_autoResize` there. Not sure if the other calls to `_autoResize` suffice. But if they do, we can then get rid of the `value` field in state (and use `props.value` as the source of truth)